flow/order/cancel.yaml

```diff
@@ -1,19 +1,30 @@
+# requires:
+#   flow.order_id (int)
+
 steps:
-  - say: I can help you with that, just let us know the reason you're cancelling
-    # TODO: move to bot settings
-#    composer:
-#      visibility: hide
+  - say: I can help you with that, just note that any discounts that you used will
+      be forfeited and can no longer be used.
+
+  - ask: Please let us know the reason you're cancelling
+    quick_replies:
+      - text: customer
+        result: customer
+      - text: fraud
+        result: fraud
+      - text: inventory
+        result: inventory
+      - text: declined
+        result: declined
+      - text: other
+        result: other
```

Comment:
Pick three of them and choose descriptions that a customer would understand. These enums as is are just for back-office categorization.

Reply:
1. Why would the customer need to manually cancel an order if payment was declined? Isn't this something the shop owner would do, or Shopify would do automatically?
1. Why would the customer need to manually cancel an order due to items out of stock? How would a customer know stock levels? Also sounds like a back-office option, if the shop owner realizes they're out of stock.

Looking at the [API & enum options](https://shopify.dev/docs/admin-api/rest/reference/orders/order#cancel-2021-01) again, I think the only valid enum value for a customer is the `customer` enum value. This is a customer requesting cancellation, so just automatically pick that enum value without asking the customer.

Reply:
Make sense...